frontend/src/components/common/Calendar/Calendar.tsx

```diff
@@ -1,38 +1,69 @@
+import { Divider } from "@chakra-ui/react";
 import React from "react";
+import { NavigationProps } from "react-hooks-helper";
 
 import { CheckIn } from "../../../types/CheckInTypes";
 import { Schedule } from "../../../types/SchedulingTypes";
 import CheckInInfoCard from "../../pages/AdminDashboard/components/CheckInInfoCard";
 import DropoffCard from "../../pages/Dashboard/components/DropoffCard";
+import ShiftCard from "../../pages/VolunteerDashboard/ShiftCard";
 import { WeeklyBody, WeeklyCalendar } from "./WeeklyCalendar";
 
 type CalendarProps = {
   selectedDay: Date;
   items: Schedule[] | CheckIn[];
   isAdminView: boolean;
   isCheckInView: boolean;
-  deleteCheckIn: (checkInId: string) => void;
+  isCheckInShiftView?: boolean;
+  setShiftId?: React.Dispatch<string>;
+  navigation?: NavigationProps;
+  setIsFoodRescue?: React.Dispatch<boolean>;
+  deleteCheckIn?: any;
 };
 
 const Calendar = ({
   selectedDay,
   items,
   isAdminView = false,
   isCheckInView = false,
+  isCheckInShiftView = false,
+  setShiftId,
+  navigation,
+  setIsFoodRescue,
   deleteCheckIn,
 }: CalendarProps): React.ReactElement => {
+  const getCheckInCard = (item: any, index: number) => {
+    if (isCheckInShiftView) {
+      return (
+        <>
+          <ShiftCard
+            key={JSON.stringify(item)}
+            shift={item}
+            setShiftId={setShiftId}
+            navigation={navigation}
+            isSignUp
+            setIsFoodRescue={setIsFoodRescue}
+          />
+          {index < items.length - 1 && <Divider pt="0.5rem" />}

```

Comment:
was mostly added here to allow the ability to add dividers that would not be added if it was the last element - open to other ideas not sure how clean this is